test/integration/targets/cloudwatchlogs/playbooks/roles/cloudwatchlogs_log_group_metric_filter/tasks/main.yml

```diff
@@ -0,0 +1,151 @@
+---
+- name: set up aws connection info
+  set_fact:
+    aws_connection_info: &aws_connection_info
+      aws_access_key: "{{ aws_access_key }}"
+      aws_secret_key: "{{ aws_secret_key }}"
+      security_token: "{{ security_token }}"
+      region: "{{ aws_region }}"
+  no_log: yes
+
+- name: test cloudwatch log group metric filter
+
+  block:
+    - name: create cloudwatch log group for integration test
+      cloudwatchlogs_log_group:
+        <<: *aws_connection_info
+        state: present
+        log_group_name: ansible/integrationtest
+        retention: 1
+
+    - name: check_mode set metric filter on ansible/integrationtest
+      cloudwatchlogs_log_group_metric_filter:
+        <<: *aws_connection_info
+        log_group_name: ansible/integrationtest
+        filter_name: MadeWithAnsible
+        filter_pattern: '{ ($.value = *) && ($.hostname = "box")}'
+        state: present
+        metric_transformation:
+          metric_name: box_free_space
+          metric_namespace: fluentd_metrics
+          metric_value: "$.value"
+      check_mode: yes
+      register: out
+
+    - name: check_mode state must be changed
+      assert:
+        that:
+          - out is changed
+
+    - name: set metric filter on ansible/integrationtest
+      cloudwatchlogs_log_group_metric_filter:
+        <<: *aws_connection_info
+        log_group_name: ansible/integrationtest
+        filter_name: MadeWithAnsible
+        filter_pattern: '{ ($.value = *) && ($.hostname = "box")}'
+        state: present
+        metric_transformation:
+          metric_name: box_free_space
+          metric_namespace: fluentd_metrics
+          metric_value: "$.value"
+      register: out
+
+    - name: create metric filter
+      assert:
+        that:
+          - out is changed
+
+    - name: re-set metric filter on ansible/integrationtest
+      cloudwatchlogs_log_group_metric_filter:
+        <<: *aws_connection_info
+        log_group_name: ansible/integrationtest
+        filter_name: MadeWithAnsible
+        filter_pattern: '{ ($.value = *) && ($.hostname = "box")}'
+        state: present
+        metric_transformation:
+          metric_name: box_free_space
+          metric_namespace: fluentd_metrics
+          metric_value: "$.value"
+      register: out
+
+    - name: metric filter must not change
+      assert:
+        that:
+          - out is not changed
+
+    - name: update matric transformation on ansible/integrationtest
+      cloudwatchlogs_log_group_metric_filter:
+        <<: *aws_connection_info
+        log_group_name: ansible/integrationtest
+        filter_name: MadeWithAnsible
+        filter_pattern: '{ ($.value = *) && ($.hostname = "box")}'
+        state: present
+        metric_transformation:
+          metric_name: box_free_space
+          metric_namespace: made_with_ansible
+          metric_value: "$.value"
+      register: out
+
+    - name: update metric filter
+      assert:
+        that:
+          - out is changed
```

Comment:
It would be nice to assert the return values in each of the tests, so that here we can assert that `metric_namespace` is `made_with_ansible` and not `fluentd_metrics`.